HW_4/HW04/HW04.Task5/Program.cs

```diff
@@ -0,0 +1,160 @@
+            {
+                if (str[i] == ' ' || i == str.Length - 1)
+                {
+                    if (currentLength > length)
+                    {
+                        if (i == str.Length - 1 && Char.IsLetter(str[i]))
+                        {
+                            currentLength += 1;
+                        }
+
+                        length = currentLength;
+
+                        index = currentIndex;
+                    }
+                    currentLength = 0;
+
+                    currentIndex = -1;
+                }
+                else
+                {
+                    if (currentIndex == -1)
+                    {
+                        currentIndex = i;
+                    }
+                    currentLength += 1;
+                }
+            }
+            str = str.Remove(index, length);
+
+            return str;
+        }
+
+        private static string SwapWords(string str)
+        {
+            string str2 = null;
+
+            string[] strArray = str.Split(new char[] { ' ' });
+
+            string longestWord = strArray[0];
+
+            string shortestWord = strArray[0];
+
+            for (int i = 1; i < strArray.Length; i++)
+            {
+                if (strArray[i].Length > longestWord.Length)
+                {
+                    longestWord = strArray[i];
+                }
+                if (strArray[i].Length < shortestWord.Length)
+                {
+                    shortestWord = strArray[i];
+                }
+            }
+            int indexLong = Array.IndexOf(strArray, longestWord);
+            int indexShort = Array.IndexOf(strArray, shortestWord);
+
+            strArray[indexLong] = shortestWord;
+            strArray[indexShort] = longestWord;
+
+            str2 = string.Join(" ", strArray);
+
+            return str2;
+        }
+
+        private static int CountAllSymbols(string str)
+        {
+            str = str.Replace(" ", "");
+
+            int symbolQuantity = 0;
+
+            for (int i = 0; i < str.Length; i++)
+            {
+                // checking for Russian letters and all punctuation marks
+                if (!Regex.IsMatch(str[i].ToString(), @"^[а-яёА-Я]+\p{P}+$"))
+                {
+                    symbolQuantity += 1;
+                }
+            }
+
+            return symbolQuantity;
+        }
+
+        private static int FindLongest(string[] strArray)
+        {
+            string longest = strArray[0];
+            int longestIndex = 0;
+            for (int i = 1; i < strArray.Length; i++)
+            {
+                if (strArray[i].Length > longest.Length)
+                {
+                    longest = strArray[i];
+                    longestIndex = i;
+                }
+            }
+            return longestIndex;
+        }
+
+        private static string[] SelectionSort(string[] strArray)
+        {
+            string[] newArray = strArray;
```

Comment:
now you have two variables pointed to one Array object in HEAP memory - b/c the Array - reference type in C#
so, that means when you change newArray - srtArray also changed and visa versa.